txrelayer/txrelayer.go

```diff
@@ -0,0 +1,189 @@
+package txrelayer
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/jsonrpc"
+	"github.com/umbracle/ethgo/wallet"
+)
+
+const (
+	defaultGasPrice = 1879048192 // 0x70000000
+	defaultGasLimit = 5242880    // 0x500000
+)
+
+var (
+	errNoAccounts               = errors.New("no accounts registered")
+	errNonSignedTxnNotSupported = errors.New("sending non signed transactions supported only for local account mode")
+)
+
+type TxRelayer interface {
+	// Call executes a message call immediately without creating a transaction on the blockchain
+	Call(from ethgo.Address, to ethgo.Address, input []byte) (string, error)
+	// SendTransaction signs given transaction by provided key and sends it to the blockchain
+	SendTransaction(transaction *ethgo.Transaction, key ethgo.Key) (*ethgo.Receipt, error)
+	// GetNonce queries nonce for the provided account
+	GetNonce(address ethgo.Address) (uint64, error)
+}
+
+type TxRelayerImpl struct {
+	client       *jsonrpc.Client
+	localAccount bool
+}
+
+func NewTxRelayer(ipAddr string, opts ...TxRelayerOption) (*TxRelayerImpl, error) {
+	client, err := jsonrpc.NewClient(ipAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	t := &TxRelayerImpl{client: client}
+	for _, opt := range opts {
+		opt(t)
+	}
+
+	return t, nil
+}
+
+func NewTxRelayerWithClient(client *jsonrpc.Client, opts ...TxRelayerOption) *TxRelayerImpl {
+	t := &TxRelayerImpl{client: client}
+	for _, opt := range opts {
+		opt(t)
+	}
+
+	return t
+}
+
+// Call executes a message call immediately without creating a transaction on the blockchain
+func (t *TxRelayerImpl) Call(from ethgo.Address, to ethgo.Address, input []byte) (string, error) {
+	callMsg := &ethgo.CallMsg{
+		From:     from,
+		To:       &to,
+		Data:     input,
+		GasPrice: defaultGasPrice,
+		Gas:      big.NewInt(defaultGasLimit),
+	}
+
+	return t.client.Eth().Call(callMsg, ethgo.Pending)
+}
+
+// SendTransaction signs given transaction by provided key and sends it to the blockchain
+func (t *TxRelayerImpl) SendTransaction(txn *ethgo.Transaction, key ethgo.Key) (*ethgo.Receipt, error) {
+	var (
+		txnHash ethgo.Hash
+		err     error
+	)
+
+	if !t.localAccount {
+		txnHash, err = t.sendSignedTransaction(txn, key)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		txnHash, err = t.sendNonSignedTransaction(txn)
```

Comment:
Split this into two functions, it is confusing in the `SendTransaction` API call that the `key` field is not used.

Besides, this is only required for funding the main account, once we find a way to do that without the local account of the Geth node (which is possible), this functionality will be removed since it is quite legacy in Geth at this point.

Reply:
Fixed [fb9abbe](https://github.com/0xPolygon/polygon-edge/pull/989/commits/fb9abbe8661f3f160cba5dd828b5fc3b14e542bf)